install-deps.sh

```diff
@@ -146,21 +146,34 @@ function install_pkg_on_ubuntu {
     fi
 }
 
+boost_ver=1.82
+boost_installed_ver=$(apt -qq list --installed 'ceph-libboost*-dev' 2> /dev/null \
+			  | grep -e 'libboost[0-9].[0-9]\+-dev' \
+			  | cut -d ' ' -f 2 \
+			  | cut -d '.' -f1,2)
+
+function clean_boost_on_ubuntu {
+    local ver=$boost_ver
+    local installed_ver=$boost_installed_ver
+    $SUDO rm -f "/etc/apt/sources.list.d/ceph-libboost${installed_ver}.list"
+    $SUDO rm -f "/etc/apt/sources.list.d/ceph-libboost.list"
+    $SUDO rm -f "/etc/apt/sources.list.d/libboost.list"
```

Comment:
> As the sources.list.d being removed in the original cleanup code isn't the one we're currently installing in the install code, add a removal for the currently used source [...]

Perhaps forestalling another set of questions from Kefu, I'm confused by this explanation.  The original cleanup code removes the version that is being installed, not the one that is intended to be installed -- it removes `ceph-libboost.*${installed_ver}.*` which is very similar to the new code removing `ceph-libboost${installed_ver}.list`.

- Under what conditions would `ceph-libboost.*${installed_ver}.*` not match?  What would the mismatch be exactly?
- What installs `ceph-libboost.list` file?
- What installs `libboost.list` file (it doesn't even have `ceph-` prefix)?

Reply:
> > As the sources.list.d being removed in the original cleanup code isn't the one we're currently installing in the install code, add a removal for the currently used source [...]
> 
> Perhaps forestalling another set of questions from Kefu, I'm confused by this explanation. The original cleanup code removes the file for the version that is already installed -- it removes `ceph-libboost.*${installed_ver}.*` which is very similar to the new code removing `ceph-libboost${installed_ver}.list`.
> 
>     * Under what conditions would `ceph-libboost.*${installed_ver}.*` not match?  What would the mismatch be exactly?

I don't know. I'm not certain that we couldn't have two ceph-libboost.*-dev packages installed and since the idea was to clean things up so it wouldn't pull in the wrong thing, I didn't think it was safe to assume there was exactly one installed version.
 
>     * What installs `ceph-libboost.list` file?

Nothing now. I removed that removal.
 
>     * What installs `libboost.list` file (it doesn't even have `ceph-` prefix)?

Line 143 in `install_pkg_on_ubuntu()` with project set to `libboost` as it is in `install_boost_on_ubuntu()`.